daemon part of the sink structure is not initialzaed when forwarding stdin to all ranks

diff -r 490e6afa37fe orte/mca/iof/hnp/iof_hnp.c
--- a/orte/mca/iof/hnp/iof_hnp.c	Tue Mar 06 11:56:15 2012 +0100
+++ b/orte/mca/iof/hnp/iof_hnp.c	Tue Mar 06 12:43:44 2012 +0100
@@ -263,6 +263,8 @@ static int hnp_push(const orte_process_n
         ORTE_IOF_SINK_DEFINE(&sink, dst_name, -1, ORTE_IOF_STDIN,
                              stdin_write_handler,
                              &mca_iof_hnp_component.sinks);
+        sink->daemon.jobid = ORTE_PROC_MY_NAME->jobid;
+        sink->daemon.vpid = ORTE_VPID_WILDCARD;
     } else {
         /* no - lookup the proc's daemon and set that into sink */
         if (NULL == (jdata = orte_get_job_data_object(dst_name->jobid))) {

